drivers/regulator/regulator_mspm0_vref.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2025 Linumiz GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_mspm0_vref
+
+#include <errno.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/dt-bindings/clock/mspm0_clock.h>
+#include <zephyr/dt-bindings/regulator/mspm0_vref.h>
+#include <zephyr/logging/log.h>
+
+/* TI Driverlib includes */
+#include <ti/driverlib/dl_vref.h>
+
+LOG_MODULE_REGISTER(vref, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define VREF_1_4V	1400000
+#define VREF_2_5V	2500000
+
+struct regulator_mspm0_vref_data {
+	struct regulator_common_data common;
+	DL_VREF_Config vref_cfg;
+};
+
+struct regulator_mspm0_vref_config {
+	struct regulator_common_config common;
+	const struct pinctrl_dev_config *vref_pin;
+	DL_VREF_ClockConfig vref_clock_cfg;
+	VREF_Regs *regs;
+};
+
+static int regulator_mspm0_vref_enable(const struct device *dev)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_enableInternalRef(config->regs);
+
+	return 0;
+}
+
+static int regulator_mspm0_vref_disable(const struct device *dev)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_disableInternalRef(config->regs);
+
+	return 0;
+}
+
+static int regulator_mspm0_vref_get_voltage(const struct device *dev, int32_t *volt_uv)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+	struct regulator_common_data *common = dev->data;
+
+	if (volt_uv == NULL) {
+		return -EINVAL;
+	}
+
+#ifdef CONFIG_REGULATOR_THREAD_SAFE_REFCNT
+	k_mutex_lock(&common->lock, K_FOREVER);
+#endif
+
+	if (config->regs->CTL0 & VREF_CTL0_BUFCONFIG_MASK) {
+		*volt_uv = VREF_1_4V;
+	} else {
+		*volt_uv = VREF_2_5V;
+	}
+
+#ifdef CONFIG_REGULATOR_THREAD_SAFE_REFCNT
+	k_mutex_unlock(&common->lock);
+#endif
+
+	return 0;
+}
+
+static int regulator_mspm0_vref_set_voltage(const struct device *dev, int32_t min_uv,
+					    int32_t max_uv)
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+	struct regulator_mspm0_vref_data *data = dev->data;
+	struct regulator_common_data *common = dev->data;
+	int32_t volt_set, volt_get;
+	int ret = 0;
+
+	if (min_uv <= VREF_2_5V && max_uv >= VREF_2_5V) {
+		volt_set = VREF_2_5V;
+	} else if (min_uv <= VREF_1_4V && max_uv >= VREF_1_4V) {
+		volt_set = VREF_1_4V;
+	} else {
+		LOG_ERR("Invalid voltage range !!");
+		return -EINVAL;
+	}
+
+#ifdef CONFIG_REGULATOR_THREAD_SAFE_REFCNT
+	k_mutex_lock(&common->lock, K_FOREVER);
+#endif
+
+	if (common->refcnt != 0) {
+		volt_get = (config->regs->CTL0 & VREF_CTL0_BUFCONFIG_MASK) ? VREF_1_4V : VREF_2_5V;
+
```

Comment:
you can remove this empty line  as your are testing below the assignment you made above.